internal/graph/reachableresources.go

```diff
@@ -86,17 +84,13 @@ func (crr *ConcurrentReachableResources) ReachableResources(
 		return err
 	}
 
-	cancelCtx, checkCancel := context.WithCancel(ctx)
-	defer checkCancel()
-
-	g, subCtx := errgroup.WithContext(cancelCtx)
-	g.SetLimit(int(crr.concurrencyLimit))
```

Comment:
so from reading `errgroup` docs, `Go` function is blocking when the concurrency limit is hit. Is the assumption that the deadlock is occurring in the `lookupRelationEntrypoint`, `lookupTTUEntrypoint` and `chunkedRedispatch`?

Reply:
One of them, yes. The test with limit 1 was able to reproduce consistently.